core/tests/protractor_desktop/topicAndStoryViewer.js

```diff
@@ -161,7 +161,7 @@ describe('Topic and Story viewer functionality', function() {
     await topicAndStoryViewerPage.goToChapterIndex(0);
     await explorationPlayerPage.submitAnswer('Continue', null);
 
-    await topicAndStoryViewerPage.login(
+    await users.createAndLoginUser(
```

Comment:
I can confirm that this change is causing the issues in the topicAndStoryViewer e2e test. What is happening here is that the user is created and logged in, and then they land on their home page (the learner dashboard). The subsequent `await` makes no sense in this context as it assumes that the learner is on the exploration player page.

Reply:
@brianrodri Can you please reply to this comment with an explanation of what actually happened?

Reply:
This is testing that you can pick up an exploration from where you left off after leaving an exploration to sign up.

The name `login` is disingenuous here, an account is being created as well. I've decided to replaced it with `createAndLoginUser` instead to make this clear.

As for the failure, I've found that the reason was that the return_url was not working as expected. This is because I was not interpolating the value correctly (comment). After fixing the interpolation, I tested locally and confirm that the redirect works.